build_docs.py

```diff
@@ -657,10 +657,25 @@ def build(self):
                 f"-D ogp_site_url={site_url}",
             )
 
+            def is_gnu_sed() -> bool:
+                """Check if we are using GNU sed."""
+                try:
+                    subprocess.run(
+                        ["sed", "--version"],
+                        stdout=subprocess.DEVNULL,
+                        stderr=subprocess.DEVNULL,
+                        check=True,
+                    )
+                    return True
+                except subprocess.CalledProcessError:
+                    return False
+                except FileNotFoundError:
+                    return False
```

Comment:
This could be a module-level function?

Reply:
Could be! This is the only place its used, so I think it's okay here, but happy to move if you like.